alphagov/paas-cf
# [#150176916] Improve API availability tests reporting

## What

This improves the reporting so that it prints a line immediately for every test failure with timestamps and the error message. It also includes the task rate in the final summary so that we have data if we need to reduce this in future.

I've taken the opportunity to do some refactoring while working on this. I've brought the app and API availability tests into subpackages under the availability namespace. This enables them to share the concourse helpers, and vendored dependencies.

## How to review

Code review is probably easiest commit-by-commit.

* Run the pipeline against an existing deploy from this branch.
* Once the api-availability tests have started, rename their test app:
```sh
cf target -o admin -s healthchecks
cf rename api-availability-test-app api-availability-test-app-hide
```
* observe test failures being reported.
* rename it back:
```sh
cf rename api-availability-test-app-hide api-availability-test-app
```

* observe the task rage being included in the final summary.

## Who can review

Not me.
